index.bs

```diff
@@ -1736,6 +2067,192 @@ The <dfn method for=StorageManager>getDirectory()</dfn> method steps are:
+FileSystemObserverObserveOptions |options|)</dfn> steps are:
+
+1. Let |result| be [=a new promise=].
+
+1. Let |recursive| be |options|["{{FileSystemObserverObserveOptions/recursive}}"].
+1. Let |observationsMap| be [=this=]'s [=FileSystemObserver/observations=].
+1. Let |locator| be |handle|'s [=FileSystemHandle/locator=].
+1. Let |global| be [=this=]'s [=relevant global object=].
+
+1. [=Enqueue the following steps=] to the [=file system queue=]:
+  1. If |observationsMap|[|locator|] [=map/exists=], abort these steps.
+
+  1. Let |entry| be the result of [=locating an entry=] given |locator|.
+  1. Let |accessResult| be the result of running |entry|'s
+     [=file system entry/query access=] given "`read`".
+
+  1. [=Queue a storage task=] with |global| to run these steps:
+    1. If |accessResult|'s [=file system access result/permission state=]
+       is not "{{PermissionState/granted}}", [=/reject=] |result| with a
+       {{DOMException}} of |accessResult|'s
+       [=file system access result/error name=] and abort these steps.
+
+    1. If |entry| is null, [=/reject=] |result| with a
+       "{{NotFoundError}}" {{DOMException}} and abort these steps.
+    1. [=Assert=]: |entry| is a [=file entry=].
+
+    1. [=Enqueue the following steps=] to the [=file system queue=]:
+      1. [=file system observation/Create an observation=] for [=this=] on |handle| with |recursive|.
+
+    1.  [=/Resolve=] |result| with null.
+
+</div>
+
+
+### The {{FileSystemObserver/unobserve()}} method ### {#api-filesystemobserver-unobserve}
+
+<div algorithm>
+The <dfn method for=FileSystemObserver>unobserve(FileSystemHandle |handle|)</dfn>
+steps are:
+
+1. Let |locator| be |handle|'s [=FileSystemHandle/locator=].
+1. Let |observationsMap| be [=this=]'s [=FileSystemObserver/observations=].
+
+1. [=Enqueue the following steps=] to the [=file system queue=]:
+  1. If |observationsMap|[|locator|] does not [=map/exist=], abort these steps.
+  1. [=file system observation/Destroy observation=] |observationsMap|[|locator|].
+
+</div>
+
+### The {{FileSystemObserver/disconnect()}} method ### {#api-filesystemobserver-disconnect}
+
+<div algorithm>
+The <dfn method for=FileSystemObserver>disconnect()</dfn> steps are:
+
+1. Let |observationsMap| be [=this=]'s [=FileSystemObserver/observations=].
+
+1. [=Enqueue the following steps=] to the [=file system queue=]:
+  1. Let |observations| be the result of [=map/getting the values=] of |observationsMap|.
+  1. [=list/For each=] |observation| in |observations|:
+    1. [=file system observation/Destroy observation=] |observation|.
+
+</div>
+
+## The {{FileSystemChangeRecord}} interface ## {#api-filesystemchangerecord}
+
+<xmp class=idl>
+enum FileSystemChangeType {
+  "appeared",
+  "disappeared",
+  "errored",
+  "modified",
+  "moved",
+  "unknown",
+};
+
+[
+    Exposed=(DedicatedWorker,SharedWorker,Window),
+    SecureContext,
+    RuntimeEnabled=FileSystemObserver
+] interface FileSystemChangeRecord {
```

Comment:
I wonder why this is an interface? It seems this could just as well be a dictionary, which might make some things simpler? If this does have to be an interface for some reason, it at least needs a constructor per https://www.w3.org/TR/design-principles/#constructors.